docs/import/add_disaster.js

```diff
@@ -60,14 +76,235 @@ function enableWhenReady() {
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function withText(td, layer, property) {
+  return td.html(layer[property]);
+}
+
+/**
+ * Add a sample of info from a list to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function withList(td, layer, property) {
+  return td.html(layer[property][0]);
+}
+
+const layerTypeStrings = new Map();
+for (const t in LayerType) {
+  if (LayerType.hasOwnProperty(t)) {
+    layerTypeStrings.set(LayerType[t], t);
+  }
+}
+
+/**
+ * Add layer type info to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+}
+ */
+function withType(td, layer, property) {
+  return td.html(layerTypeStrings.get((layer[property])));
+}
+
+/**
+ * Add checkbox capabilities to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLElement> }
+ */
+function withCheckbox(td, layer, property) {
+  const checkbox = $(document.createElement('input'))
+                       .prop('type', 'checkbox')
+                       .prop('checked', layer[property])
+                       .on('change', (event) => {
+                         const index = $('#tbody > tr').length -
+                             $('tr').index($(event.target).parents('tr'));
+                         getCurrentLayers()[index][property] =
+                             checkbox.is(':checked');
+                         return updateLayersInFirestore();
+                       });
+  return $(td).append(checkbox);
+}
+
+/**
+ * Create an instance of the color boxes for the color col.
+ * @param {string} color what color to make the box.
+ * @return {JQuery<HTMLDivElement>}
+ */
+function createColorBox(color) {
+  return $(document.createElement('div'))
+      .addClass('box')
+      .css('background-color', color);
+}
+
+/**
+ * Add color function info to the given td.
+ * @param {JQuery<HTMLElement>} td
+ * @param {Object} layer
+ * @param {string} property
+ * @param {number} index
+ * @return {JQuery<HTMLElement>}
+ */
+function withColor(td, layer, property, index) {
+  const colorFunction = layer[property];
+  if (!colorFunction) {
+    td.html('N/A').addClass('na');
+  } else if (colorFunction['single-color']) {
+    td.append(createColorBox(colorFunction['single-color']));
+  } else if (colorFunction['base-color']) {
+    td.append(createColorBox(colorFunction['base-color']));
+  } else if (colorFunction['colors']) {
+    const colorObject = colorFunction['colors'];
+    const colorSet = new Set();
+    Object.keys(colorObject).forEach((propertyValue) => {
+      const color = colorObject[propertyValue];
+      if (!colorSet.has(color)) {
```

Comment:
nit: Save a line by doing if (colorSet.add(color)) {...}, since add returns whether or not the add happened.

Reply:
ugh, the worst. my bad